### ISSUE TITLE
IPFS js cant connect to ws://127.0.0.1:8081/p2p/

### ISSUE THREAD

**Issue by adamprocter** [ipfs/js-ipfs#3298]
Wanted to add as issue as discussed on stackoverflow (see link) as still in latest version.

- **Version**:
0.50.2

- **Platform**:
Firefox / Safari

#### Severity:
Low - Annoying console logging of errors, likely to effect browser / web app performance

#### Description:
Using IPFS I get the errors of either not allowed to run insecure content from ws://127.0.0.1:8081/p2p/Qmbut9Ywz9YEDrz8ySBSgWyJk41Uvm2QJPhwDJzJyGFsD6.

or Firefox can’t establish a connection to the server at ws://127.0.0.1:8081/p2p/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64.

Here is more on the issue and what appears to be a note towards fix
https://stackoverflow.com/questions/63563162/ipfs-js-cant-connect-to-ws-127-0-0-18081-p2p/63612116#63612116

**Comment by achingbrain** [comment]
This is caused by network peers announcing local addresses as being externally dialable.  It's annoying but safe to ignore the printed errors.

I believe the next release of libp2p will not log these errors so the problem should go away.

cc: @vasco-santos

**Comment by jacobheun** [comment]
> I believe the next release of libp2p will not log these errors so the problem should go away.

The logging can't be stopped, it's a security log in browsers that's purposefully uncatchable. We are looking at adding the IP/port combo to a denylist to prevent subsequent/frequent dials to those to minimize the number of errors being logged.

**Comment by adamprocter** [comment]
do we have any progress on this ? Thanks

**Comment by vasco-santos** [comment]
Yes, this will be fixed by this: https://github.com/libp2p/js-libp2p-websockets/pull/116

As this is a breaking change, it will be included in `js-libp2p@0.30` release https://github.com/libp2p/js-libp2p/issues/655 . Once it is ready, we can bubble it up to IPFS

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).

Please feel to reopen with any comments before 2023-06-05. We will do a final pass on reopened issues afterward (see https://github.com/ipfs/js-ipfs/issues/4336).
